BrowserKit/Sources/SiteImageView/SiteImageHandler.swift

```diff
@@ -28,38 +25,22 @@ public class DefaultSiteImageHandler: SiteImageHandler {
         self.imageHandler = imageHandler
     }
 
-    public func getImage(urlStringRequest: String,
-                         type: SiteImageType,
-                         id: UUID,
-                         usesIndirectDomain: Bool) async -> SiteImageModel {
-        var imageModel = SiteImageModel(id: id,
-                                        expectedImageType: type,
-                                        urlStringRequest: urlStringRequest,
-                                        siteURL: nil,
-                                        cacheKey: "",
-                                        domain: nil,
-                                        faviconURL: nil,
-                                        faviconImage: nil,
-                                        heroImage: nil)
+    public func getImage(site: SiteImageModel) async -> SiteImageModel {
+        var imageModel = site
 
         // urlStringRequest possibly cannot be a URL
-        if let siteURL = URL(string: urlStringRequest) {
+        if let siteURL = URL(string: site.siteURLString ?? "") {
```

Comment:
would it have any failure points if we mark it an empty string?

Reply:
An empty string will fail to convert to a URL which means it will eventually fallback to use a generated letter image or in the case of an empty string the default globe image.